scripts/dts/extract_dts_includes.py

```diff
@@ -515,6 +515,11 @@ def dict_merge(parent, fname, dct, merge_dct):
             dict_merge(k, fname, dct[k], merge_dct[k])
         else:
             if k in dct and dct[k] != merge_dct[k]:
+                # Don't warn if we are changing the "category" from "optional
+                # to "required"
+                if (k == "category" and dct[k] == "optional" and
+                    merge_dct[k] == "required"):
+                    continue
```

Comment:
I see this eliminates the warning, but doesn't the `continue` cause it to stop processing the iteration before it updates `dct[k]` with the new (stricter) condition?  IOW it doesn't overwrite, instead keeping `optional`.

Reply:
Good point, I reworked how the logic works.